internal/core/hls_muxer.go

```diff
@@ -296,6 +298,13 @@ func (m *hlsMuxer) runInner(innerCtx context.Context, innerReady chan struct{})
 			"the stream doesn't contain any supported codec (which are currently H264, H265, MPEG4-Audio, Opus)")
 	}
 
+	var muxerDirectory string
+	if m.directory != "" {
+		muxerDirectory = filepath.Join(m.directory, m.pathName)
+		os.MkdirAll(muxerDirectory, 0o755)
+		defer os.Remove(muxerDirectory)
```

Comment:
it is possible to set up this function to allow chunks to be kept even if the muxer goes wrong?, which can happen since many cameras do not strictly comply with standards.

Reply:
@andrew-ld we can surely setup a failback mechanism, i'll reply to your issues and PR ASAP.